app/models/application_record.rb

```diff
@@ -16,6 +16,15 @@ def self.needs_migration?
     ActiveRecord::Migration.check_pending!
   end
 
+  # Determine whether the given search term is possibly a Primary Key (it's numeric and less than 4 bytes)
+  def self.possibly_pk?(search_term) # could add optional arg for 4 byte vs 8 byte, if needed later
```

Comment:
I don't think this should go in ApplicationRecord. That seems like a very general bucket as every model in the app inherits from it. Also our tables have inconsistencies around pks so this may not be the correct logic for every model.

Since it's so simple I'd be tempted to either inline it into the scopes where it's used or drop it in a bag of utility functions like `HudUtility.term_is_possibily_pk?(term)`

Reply:
How about `Hmis::Hud::Base` or `Hmis::Hud::Concerns::Shared`?  This appears to be something we'd only use in the `Hud` namespace, and isn't used on the warehouse side right?

Reply:
It is used on the warehouse side because the client search concern uses it.Or we could just inline that specific use-case.

Reply:
Thanks then I _think_ `HmisStructure::Base` fits the bill.

Reply:
Based on the proposal to leave the client search concern unchanged, per the thread above, I've moved it to `Hmis::Hud::Base`.